app/models/form1095_b.rb

```diff
@@ -0,0 +1,137 @@
+      Rails.logger.error "1095-B template for year #{tax_year} does not exist."
+      raise "1095-B for tax year #{tax_year} not supported"
+    end
+
+    pdf.fill_form(
+      pdf_template,
+      temp_file,
+      {
+        "topmostSubform[0].Page1[0].Pg1Header[0].cb_1[1]": data['is_corrected'] && 2,
+        "topmostSubform[0].Page1[0].Part1Contents[0].Line1[0].f1_01[0]": data['first_name'],
+        "topmostSubform[0].Page1[0].Part1Contents[0].Line1[0].f1_02[0]": data['middle_name'],
+        "topmostSubform[0].Page1[0].Part1Contents[0].Line1[0].f1_03[0]": data['last_name'],
+        "topmostSubform[0].Page1[0].Part1Contents[0].f1_04[0]": data['ssn'] || '',
+        "topmostSubform[0].Page1[0].Part1Contents[0].f1_05[0]": data['ssn'] ? '' : data['birth_date'],
+        "topmostSubform[0].Page1[0].Part1Contents[0].f1_06[0]": data['address'],
+        "topmostSubform[0].Page1[0].Part1Contents[0].f1_07[0]": data['city'],
+        "topmostSubform[0].Page1[0].Part1Contents[0].f1_08[0]": data['state'] || data['province'],
+        "topmostSubform[0].Page1[0].Part1Contents[0].f1_09[0]": country_and_zip,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].f1_25[0]": data['first_name'],
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].f1_26[0]": middle_initial,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].f1_27[0]": data['last_name'],
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].f1_28[0]": data['ssn'] || '',
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].f1_29[0]": data['ssn'] ? '' : data['birth_date'],
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_01[0]": data['coverage_months'][0] && 1,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_02[0]": data['coverage_months'][1] && 1,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_03[0]": data['coverage_months'][2] && 1,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_04[0]": data['coverage_months'][3] && 1,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_05[0]": data['coverage_months'][4] && 1,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_06[0]": data['coverage_months'][5] && 1,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_07[0]": data['coverage_months'][6] && 1,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_08[0]": data['coverage_months'][7] && 1,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_09[0]": data['coverage_months'][8] && 1,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_10[0]": data['coverage_months'][9] && 1,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_11[0]": data['coverage_months'][10] && 1,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_12[0]": data['coverage_months'][11] && 1,
+        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_13[0]": data['coverage_months'][12] && 1
```

Comment:
What are you trying to do here? Are you trying to convert a boolean value to an integer? `true=1`, `false=0` ? If the data value is false `false && 1` will return `false`. If you want false to return 0 you will need to do `... && 1 || 0`. Can you add some specs to test when there are instances of false?
```suggestion
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_01[0]": data['coverage_months'][0] && 1 || 0,
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_02[0]": data['coverage_months'][1] && 1 || 0,
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_03[0]": data['coverage_months'][2] && 1 || 0,
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_04[0]": data['coverage_months'][3] && 1 || 0,
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_05[0]": data['coverage_months'][4] && 1 || 0,
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_06[0]": data['coverage_months'][5] && 1 || 0,
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_07[0]": data['coverage_months'][6] && 1 || 0,
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_08[0]": data['coverage_months'][7] && 1 || 0,
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_09[0]": data['coverage_months'][8] && 1 || 0,
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_10[0]": data['coverage_months'][9] && 1 || 0,
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_11[0]": data['coverage_months'][10] && 1 || 0,
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_12[0]": data['coverage_months'][11] && 1 || 0,
        "topmostSubform[0].Page1[0].Table1_Part4[0].Row23[0].c1_13[0]": data['coverage_months'][12] && 1 || 0
```

Reply:
This is due to pdf_fill being very strange. It doesn't like the boolean value of true for checking the box in the pdf, it likes 1. A false value doesn't check the box, which is what we want. So logically here, 'boolean && 1' serves its purpose, which is to have false if false, or 1 if true